macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +271,355 @@ open class Button: NSButton {
-				height = imageSize.height
-				break
-			case .imageBelow:
-				x = (rect.width - imageSize.width) / 2
-				y = Button.verticalEdgePadding + titleSize.height + Button.titleAndImageRectInterspacing
-				width = imageSize.width
-				height = imageSize.height
-				break
-			case .imageAbove:
-				x = (rect.width - imageSize.width) / 2
-				y = Button.verticalEdgePadding
-				width = imageSize.width
-				height = imageSize.height
-				break
-			case .imageOverlaps:
-				x = Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = rect.width - (Button.horizontalEdgePadding * 2)
-				height = rect.height - (Button.verticalEdgePadding * 2)
-				break;
-			case .imageLeading:
-				x = isRTL ? titleSize.width + Button.titleAndImageRectInterspacing + Button.horizontalEdgePadding : Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = imageSize.width
-				height = imageSize.height
-				break
-			case .imageTrailing:
-				x = isRTL ? Button.horizontalEdgePadding : titleSize.width + Button.titleAndImageRectInterspacing + Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = imageSize.width
-				height = imageSize.height
+	/// This color is used for the background in the primary style, the pressed background in the secondary
+	/// style, and the content tint (i.e. foreground text/image) color in the borderless style.  It is not used
+	/// for the acrylic style.
+	@objc public var accentColor: NSColor? = Colors.primary
+
+	/// Any of several pre-set button styles.  Setting this property to `.none` does nothing.  Setting it to
+	/// any other value determines the content tint (i.e. foreground text/image), background and border
+	/// color properties for all button states, discarding any previous values.
+	@objc open var style: ButtonStyle = .primary {
+		willSet {
+			switch newValue {
+			case .primary:
+				contentTintColor = ButtonColor.neutralInverted
+				contentTintColorDisabled = ButtonColor.brandForegroundDisabled
+				backgroundColor = accentColor
+				backgroundColorDisabled = ButtonColor.brandBackgroundDisabled
+				borderColor = .clear
+			case .secondary:
+				contentTintColor = .textColor
+				contentTintColorPressed = ButtonColor.neutralInverted?.withSystemEffect(.pressed)
+				backgroundColor = ButtonColor.neutralBackground2
+				backgroundColorPressed = accentColor?.withSystemEffect(.pressed)
+				borderColor = ButtonColor.neutralStroke2
+				borderColorPressed = .clear
+			case .acrylic:
+				contentTintColor = ButtonColor.neutralForeground3
+				backgroundColor = ButtonColor.neutralBackground3
+				borderColor = .clear
+			case .borderless:
+				contentTintColor = accentColor
+				contentTintColorDisabled = ButtonColor.brandForegroundDisabled
+				backgroundColor = .clear
+				borderColor = .clear
+			case .none:
 				break
 			@unknown default:
 				break
 			}
+		}
+	}
 
-			return NSMakeRect(x, y, width, height)
+	private var fontSize: CGFloat = 15 {
```

Comment:
Nit, move magic number to a constant

Reply:
If these all need to be separately defined constants then I would like to organize them into a struct.  But I need to understand the comments about avoiding structs first.

Reply:
Fair. I personally wouldn't block a review on it, but the current guidance we as APEX has chosen is fileprivate constants at the bottom of the file, as seen here https://microsoft.github.io/swift-guide/FileContentOrdering.html

I don't quite like `fileprivate` but I did keep my constants at the bottom to follow this.